project/Build.scala

```diff
@@ -67,9 +68,18 @@ object DottyBuild extends Build {
     // otherwise it just executes the tests directly
     lockPartestFile := {
       val partestLockFile = "." + File.separator + "tests" + File.separator + "partest.lock"
-      partestLock = new RandomAccessFile(partestLockFile, "rw").getChannel.tryLock
+      try {
+        partestLock = new RandomAccessFile(partestLockFile, "rw").getChannel.tryLock
+      } catch {
+        case ex: java.nio.channels.OverlappingFileLockException => // locked already
```

Comment:
I added exceptions in those two cases (commit below), however running `test` while a concurrent `partest` is running will not throw an exception and result in the undesired behavior described above, and depending on the timing other interleavings of test & partest could mess things up as well... So generally one should avoid running tests in two concurrent sbt instances.